File: selenium-java/src/main/java/pages/CartPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class CartPage {

    private final WebDriver driver;
    private final WebDriverWait wait;
    private final By checkoutButton = By.cssSelector("[data-test='checkout']");
    private final By cartItems = By.cssSelector(".cart_item");

    public CartPage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public void proceedToCheckout() {
        wait.until(ExpectedConditions.elementToBeClickable(checkoutButton)).click();
    }

    public int getCartItemCount() {
        return driver.findElements(cartItems).size();
    }
}

File: pytest-selenium/pages/cart_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webdriver import WebDriver


class CartPage:
    def __init__(self, driver: WebDriver):
        self.driver = driver
        self._checkout_button = (By.CSS_SELECTOR, '[data-test="checkout"]')
        self._cart_items = (By.CSS_SELECTOR, ".cart_item")

    def get_cart_item_count(self) -> int:
        return len(self.driver.find_elements(*self._cart_items))

    def proceed_to_checkout(self):
        self.driver.find_element(*self._checkout_button).click()
